catalog/plugin/catalog-plugin-auditclassifiedaccess/src/main/java/org/codice/alliance/catalog/plugin/auditclassified/AuditClassifiedAccessPlugin.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.catalog.plugin.auditclassified;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.common.audit.SecurityLogger;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuditClassifiedAccessPlugin implements PostQueryPlugin {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AuditClassifiedAccessPlugin.class);
+
+  private List<String> enteredClassificationValues = new ArrayList<>();
+
+  private List<String> enteredReleasabilityValues = new ArrayList<>();
+
+  private List<String> enteredDisseminationControlsValues = new ArrayList<>();
+
+  private List<String> enteredCodewordsValues = new ArrayList<>();
+
+  public QueryResponse process(QueryResponse input)
+      throws PluginExecutionException, StopProcessingException {
+
+    List<Result> results = input.getResults();
+
+    results.stream().map(Result::getMetacard).filter(Objects::nonNull).forEach(this::handleAudits);
+
+    // NOTE: this log here shows up a bit, should probably be removed
+    LOGGER.info("Response went through the Audit Classified Access Plugin");
+    return input;
+  }
+
+  private void handleAudits(Metacard metacard) {
+    auditClassification(metacard);
+    auditReleasability(metacard);
+    auditDisseminationControls(metacard);
+    auditCodewords(metacard);

```

Comment:
✏️ **[Map]**: You could just iterate over the `Entry`s of the Map here and call the same generic method each time, passing in the `Entry`'s key and value in addition to the metacard. Then that generic method would do what these are all doing except they would grab the attribute based on whatever attribute key is passed in. You'll just have one method instead of these 4.